### PR TITLE
TRITON-2050 TOC generation using doctoc

### PR DESCRIPTION
**Pull request by kusor**
Adding all of you for review just to note that `make tocs` will generate a proper table of content now.
Happy to get everybody input since I'd like to do the same with the other Triton repos.

### CODE REVIEW

**Comment by trentm** [comment]
Other than ^^, it LGTM.

File: Makefile [hunk 1]
```diff
@@ -132,8 +129,16 @@ release: all deps docs $(SMF_MANIFESTS)
...
 
-regen_docs:
-	$(NODE) ./tools/gendocs.js docs/static.md lib/endpoints > docs/index.md
+./node_modules/.bin/doctoc:
+	npm install
+
+# Make a table of contents in Markdown docs that are setup to use it.  This
+# changes those files in-place, so one should do this before commit.
+.PHONY: tocs
+tocs: | ./node_modules/.bin/doctoc
```
Comment (twhiteman) [inline Makefile]: I think it should be `toc`, not the plural `tocs`, i.e. `make toc`

File: Makefile [hunk 2]
```diff
@@ -132,8 +129,16 @@ release: all deps docs $(SMF_MANIFESTS)
 	(cd $(RELSTAGEDIR) && $(TAR) -I pigz -cf $(ROOT)/$(RELEASE_TARBALL) root site)
 	@rm -rf $(RELSTAGEDIR)
 
-regen_docs:
-	$(NODE) ./tools/gendocs.js docs/static.md lib/endpoints > docs/index.md
+./node_modules/.bin/doctoc:
+	npm install
```
Comment (twhiteman) [inline Makefile]: Does this need to be the full `npm install` or can it be `npm install doctoc` ?

Comment (trentm) [inline Makefile]:
```suggestion
	$(NPM) install
```

should also be the NPM var

File: Makefile (hunk 1 above)
Comment (trentm) [reply Makefile]: Or drop the separate "toc" target and put the `doctoc ...` command(s) under the `doc` target.

File: package.json [hunk 3]
```diff
@@ -35,7 +35,8 @@
   },
   "devDependencies": {
     "ejs": "^2.4.1",
```
Comment (trentm) [inline package.json]: You can drop the "ejs" dep now that gendocs.js is gone.

Comment (kusor) [reply package.json]: Good call!

**Pull request merged**